### ISSUE TITLE
Contrib Loss Function Bug

### ISSUE THREAD

**Issue by jhorowitz** [tensorflow/tensorflow#20253]
### System information
I've reproduced this problem on multiple systems including two with CPU only and three with GPU. The specs below represent one such system.

Have I written custom code
Yes

OS Platform and Distribution
Mac OS X 10.13.2
17.3.0 Darwin Kernel 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64

TensorFlow installed from
Pip (python3)

TensorFlow version
1.8.0

Bazel version
N/A

CUDA/cuDNN version
N/A

GPU model and memory
N/A

Exact command to reproduce
Please see code below

### Describe the problem
There are several conditions that cause triplet_semihard_loss from tf.contrib to return nan as the loss. This totally messes up the gradient and causes divergence (at least in my build). Below is a minimal reproduction of one way to cause nan which is to have a batch of all different classes which causes there to be no positive for any chosen anchor. 

The lack of a negative does not result in nan, just a super high loss which while not as bad, is still not what I'd expect.

When running backprop, I would expect to have no gradient if there are no valid triplets to train off of.

I'm happy to contribute to fixes if you guys agree this is a bug.

### Source code / logs
```python
import numpy as np
import tensorflow as tf
from tensorflow.contrib.losses import metric_learning

labels = tf.placeholder(tf.int32, [None], name='labels')
embeddings = tf.placeholder(tf.float32, [None, 128], name='embeddings')

loss_ph = metric_learning.triplet_semihard_loss(labels, embeddings)

sess = tf.InteractiveSession()
sess.run(tf.global_variables_initializer())

loss = sess.run(loss_ph, feed_dict={
    embeddings: np.random.rand(2, 128),
    labels: np.array([1, 2])
})

print("loss", loss)
# loss nan
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jhorowitz** [comment]
@tensorflowbutler Updated!

**Comment by jhorowitz** [comment]
Bump @alextp 
I imagine you're pretty busy. I just want to make sure you've seen this.

**Comment by alextp** [comment]
I've seen this. However I didn't write the code nor do we promise we support code in contrib to the same level. Glancing over the implementation I'm not surprised you see NaNs.

I'd happy if you submit pull requests clearing this up somehow.

**Comment by jhorowitz** [comment]
Hi @coreylynch,

I definitely think usability could be improved via documentation. The issues exists in other scenarios as well when there are both positive and negative examples for any given anchor, lacking a positive was just the easiest to reproduce. The tf.assert would need to compute the triplets in order to comprehensively determine if a nan will be produced. 

I can provide samples if that would be helpful. Please advise.

Thanks,
Josh

**Comment by dynamicwebpaige** [comment]
Closing out this issue, as `tf.contrib` is no longer supported in TensorFlow 2.0. 

If you would like to debug the code for `tensorflow.contrib.losses.metric_learning` and upgrade the code for the loss function, a good home for it might be [TensorFlow Addons](https://www.github.com/tensorflow/addons). Thanks! 🙂
